alf/environments/wrappers.py

```diff
@@ -102,3 +107,48 @@ def step(self, action):
 
     def reset(self, **kwargs):
         return self.env.reset(**kwargs)
+
+
+@gin.configurable
+class ImageScaleTransformer(gym.Wrapper):
+    """
+    Scale uint8 image input to min and max (0->min, 255->max)
+
+    Note: it treats an observation with len(shape)==4 as image
+    Args:
+        observation (nested Tensor): observations
+        min (float): normalize minimum to this value
+        max (float): normalize maximum to this value
+    Returns:
+        Transfromed observation
+    """
+    def __init__(self, env, minmax=(-1., 1.)):
+        super().__init__(env)
+        self._minmax = minmax
+
+    def __getattr__(self, name):
+        """Forward all other calls to the base environment."""
+        return getattr(self.env, name)
+
+    def reset(self):
+        observation = self.env.reset()
+        return self._transform(observation)
+
+    def step(self, action):
+        observation, reward, done, info = self.env.step(action)
+        return self._transform(observation), reward, done, info
+
+    def _transform(self, observation):
+        def _transform_image(obs):
+            # tf_agent changes all gym.spaces.Box observation to tf.float32.
+            # See _spec_from_gym_space() in tf_agents/environments/gym_wrapper.py
+            min, max = self._minmax
+            if len(obs.shape) == 4:
+                if obs.dtype == tf.uint8:
```

Comment:
the observation is likely to be a numpy array, can its dtype be compared against tf dtypes?

Reply:
it's not correct. fixed